BuildingBlocks.Infrastructure/BuildingBlocks.Infrastructure/Persistence/BlobStorage/GoogleCloudStorage/GoogleCloudStorage.cs

```diff
@@ -62,6 +62,31 @@ public async Task<byte[]> FindAsync(string blobId)
             }
         }
 
+        public async Task<IEnumerable<string>> FindAllAsync()
+        {
+            _logger.LogTrace($"Listing all blobs...");
+
+            try
+            {
+                var storageList = new List<string>();
+                var storageObjects = _storageClient.ListObjectsAsync(_bucketName);
+
+                await foreach (var storageObject in storageObjects)
+                {
+                    storageList.Add(storageObject.Name);
+                }
```

Comment:
You can simplify this using Linq:

```suggestion
                var blobs = await _storageClient
                    .ListObjectsAsync(_bucketName)
                    .Select(storageObject => storageObject.Name)
                    .ToListAsync();
```

(you need to install the NuGet package System.Interactive.Async for this to compile)

Reply:
If you return IAsyncEnumerable, you have to change it as follows, of course:

```suggestion
                var blobs = _storageClient
                    .ListObjectsAsync(_bucketName)
                    .Select(storageObject => storageObject.Name);
```